engine/execution/ingestion/engine.go

```diff
@@ -359,7 +359,10 @@ func (e *Engine) onBlockExecuted(executed *entity.ExecutableBlock, finalState fl
 			// find the block that was just executed
 			executionQueue, exists := executionQueues.ByID(executed.ID())
 			if !exists {
-				return fmt.Errorf("fatal error - executed block not present in execution queue")
+				// when the block no longer exists in the queue, it means there was a race condition that
+				// two onBlockExecuted was called for the same block, and one process has already removed the
+				// block from the queue, so we could simply bail here.
+				return nil
```

Comment:
This situation can only happen if we are during syncing right? That block is both executed and synced from other node.
If so, I think we should distinguish those two situations and error out if we are not syncing - because that would mean a block was executed twice (or the handler was called twice) which is not a good thing.

Reply:
From the log, we did see the block was executed twice. I believe it happens because the `executeBlockIfComplete` was triggered twice. 

How did we prevent a block from executing more than once before?

But yeah, I will change to return an error, but not fatal error.

Reply:
> From the log, we did see the block was executed twice. I believe it happens because the executeBlockIfComplete was triggered twice.

this sounds like some coding error. Maybe not fatal, but something we shouldn't just suppress.

> How did we prevent a block from executing more than once before?

We used all those `Backend` structures which were synchronized and accessed single-threaded.